gems/smithy-client/sig/smithy-client/http/request.rbs

```diff
@@ -0,0 +1,18 @@
+module Smithy
+  module Client
+    module HTTP
+      class Request
+
+        def initialize: (?Hash[Symbol, untyped]) -> void
+
+        attr_reader endpoint: endpoint_url?
+        attr_accessor http_method: String
+        attr_reader headers: Headers
+        attr_reader body:  _ReadableIO?
+
+        def body=: (_ReadableIO | String | nil) -> void
```

Comment:
Same here as above

Reply:
Ditto here on the ReadbleIO interface.  It *also* has a weird case where it will convert a String to a StringIO for you (and String ofc does not meet the ReadableIO requirements, so needs to be added explicitly to the type signature).